src/promptflow/tests/sdk_cli_test/unittests/test_utils.py

```diff
@@ -488,3 +490,43 @@ def test_gen_uuid_by_compute_info(self):
         system_info_hash = hashlib.sha256((host_name + system + machine).encode()).hexdigest()
         compute_info_hash = hashlib.sha256((mac_address + system_info_hash).encode()).hexdigest()
         assert str(uuid.uuid5(uuid.NAMESPACE_OID, compute_info_hash)) == gen_uuid_by_compute_info()
+
+    def test_executable_package_match_toml_file(self):
+        def get_git_base_dir():
+            return Path(
+                subprocess.run(["git", "rev-parse", "--show-toplevel"], stdout=subprocess.PIPE)
+                .stdout.decode("utf-8")
+                .strip()
+            )
+
+        def get_toml_dependencies():
+            packages = ["promptflow-tracing", "promptflow-core", "promptflow-devkit"]
+            dependencies = []
+
+            for package in packages:
+                with open(get_git_base_dir() / "src" / package / "pyproject.toml", "rb") as file:
+                    data = toml.load(file)
+                extra_package_names = data.get("tool", {}).get("poetry", {}).get("dependencies", {})
+                dependencies.extend(extra_package_names.keys())
+            dependencies = [
+                dependency
+                for dependency in dependencies
+                if not dependency.startswith("promptflow") and not dependency == "python"
+            ]
+            return dependencies
+
+        all_packages = get_toml_dependencies()
+
+        with open(
```

Comment:
add more comments here, so devs know what to do when they see fails

Reply:
done